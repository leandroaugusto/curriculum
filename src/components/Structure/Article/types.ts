export interface ArticleProps {
  readonly children: React.ReactNode;
  readonly company: string | React.ReactNode;
  readonly year: string | React.ReactNode;
  readonly role: string | React.ReactNode;
  readonly jobType: string | React.ReactNode;
}
