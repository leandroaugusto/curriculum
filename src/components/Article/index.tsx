interface ArticleProps {
  readonly children: React.ReactNode;
  readonly company: string;
  readonly year: string;
  readonly role: string;
}

export default function Article({
  children,
  company,
  year,
  role,
}: ArticleProps) {
  return (
    <article className="my-2">
      <h4 className="font-bold">{year}</h4>
      <h4 className="font-bold">{role}</h4>
      <h5 className="my-2 text-sm font-black text-cyan-600">{company}</h5>
      {children}
    </article>
  );
}
