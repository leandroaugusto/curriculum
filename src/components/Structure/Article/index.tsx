interface ArticleProps {
  readonly children: React.ReactNode;
  readonly company:
    | string
    | React.ReactElement<any, string | React.JSXElementConstructor<any>>
    | React.ReactNodeArray;
  readonly year:
    | string
    | React.ReactElement<any, string | React.JSXElementConstructor<any>>
    | React.ReactNodeArray;
  readonly role:
    | string
    | React.ReactElement<any, string | React.JSXElementConstructor<any>>
    | React.ReactNodeArray;
  readonly jobType:
    | string
    | React.ReactElement<any, string | React.JSXElementConstructor<any>>
    | React.ReactNodeArray;
}

export default function Article({
  children,
  company,
  year,
  role,
  jobType,
}: ArticleProps) {
  return (
    <article className="mt-4 mb-6">
      <h4 className="text-sm font-bold">
        {year} <span>| {jobType}</span>
      </h4>
      <h4 className="mt-2 font-bold text-cyan-800">{role}</h4>
      <h5 className="mb-2 text-sm font-bold text-cyan-800">{company}</h5>
      {children}
    </article>
  );
}
