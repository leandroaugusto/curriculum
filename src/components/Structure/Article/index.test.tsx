import { render } from "@testing-library/react";
import Article from ".";

describe("Article", () => {
  it("renders the Article element", () => {
    const { container } = render(
      <Article company="company" role="role" year="2010" jobType="jobType">
        <p>Description</p>
      </Article>
    );

    expect(container).toMatchSnapshot();
  });
});
