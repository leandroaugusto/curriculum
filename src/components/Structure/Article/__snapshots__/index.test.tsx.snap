// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders the Article element 1`] = `
<div>
  <article
    class="mt-4 mb-6"
  >
    <h4
      class="text-sm font-bold"
    >
      2010
       
      <span>
        | 
        jobType
      </span>
    </h4>
    <h4
      class="mt-2 font-bold text-cyan-800"
    >
      role
    </h4>
    <h5
      class="mb-2 text-sm font-bold text-cyan-800"
    >
      company
    </h5>
    <p>
      Description
    </p>
  </article>
</div>
`;
